analyzers/tests/SonarAnalyzer.UnitTest/Rules/StaticFieldWrittenFromInstanceMemberTest.cs

```diff
@@ -18,28 +18,65 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
 using SonarAnalyzer.Rules.CSharp;
 
 namespace SonarAnalyzer.UnitTest.Rules
 {
     [TestClass]
     public class StaticFieldWrittenFromInstanceMemberTest
     {
+        private readonly VerifierBuilder<StaticFieldWrittenFromInstanceMember> builder = new();
+
         [TestMethod]
         public void StaticFieldWrittenFromInstanceMember() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\StaticFieldWrittenFromInstanceMember.cs",
-                                    new StaticFieldWrittenFromInstanceMember(),
-                                    ParseOptionsHelper.FromCSharp8,
-                                    MetadataReferenceFacade.NETStandard21);
+            builder.AddPaths(@"StaticFieldWrittenFromInstanceMember.cs").WithOptions(ParseOptionsHelper.FromCSharp8).AddReferences(MetadataReferenceFacade.NETStandard21).Verify();
+
+        [TestMethod]
+        public async Task SecondaryIssueInReferencedCompilation()
+        {
+            const string firstClass =
+                @"
+public class Foo
+{
+    public static int Count = 0; // Secondary
+}
+";
+
+            const string secondClass =
+                @"
+public class Bar
+{
+    public int Increment() => Foo.Count++;
+}
+";
+
+            var analyzers = ImmutableArray<DiagnosticAnalyzer>.Empty.Add(new StaticFieldWrittenFromInstanceMember());
+            var firstCompilation = CreateCompilation(CSharpSyntaxTree.ParseText(firstClass), "First").WithAnalyzers(analyzers).Compilation;
+            var secondCompilation = CreateCompilation(CSharpSyntaxTree.ParseText(secondClass), "Second")
+                                    .AddReferences(firstCompilation.ToMetadataReference())
+                                    .WithAnalyzers(analyzers);
+
+            var result = await secondCompilation.GetAnalyzerDiagnosticsAsync();
+            result.Single().Id.Should().Be("S2696");
```

Comment:
```suggestion
            result.Should().BeEquivalent(new [] { new { Id: "S2696"} });
```

Reply:
I'm not sure I understand why this is better?

Reply:
I would say for this reasons:
* It gives a better error message if the number of elements in `result` does not match the expected number of elements
* The message for "Id" is slightly better.
* You can easily extend the test for properties of `Diagnostic`.
* It is more idiomatic for FluentAssertions.